serialize/deserialize round trip

mcshaz opened the pull request:
as per previous discussions, tests of strings are removed, and the serialize - > deserialize round trip is used. multiple more tests are added, all of them now passing (214 tests in total passing). 

This request includes multiple commits, and you are welcome to tell me which you would/would not like.

I now realize should have merged the remote sooner in order that you didn't have to see your older code (apologies, I am learning git). Needless to say all the changes you have made in the past week have been included.

As per the titles includes:
- posting a web request to an online validator - tenuous, but of some use at the moment (I have reached out to them to provide JSON output, but no response yet). This has helped me find heaps of things not meeting spec.
- the timezone serialization as per discussion/spec, many comments for the developer
- bug fix for datetime UTC midnight only showing date

mcshaz committed:
Merge remote-tracking branch 'refs/remotes/rianjs/master'

mcshaz committed:
whitespace begin folded line

mcshaz committed:
dictionary  TryGetValue

mcshaz committed:
EnumSerializer converting ToUpper

mcshaz committed:
CamelCase to hyphenated

mcshaz committed:
spec validation

mcshaz committed:
Serialization Tests now object -> serialised -> deserialised

mcshaz committed:
More TimeZone tests

mcshaz committed:
Validating to spec

mcshaz committed:
broken test comments updated

mcshaz committed:
remove VTimeZone includeHistoricaData argument

mcshaz committed:
minor debugging clarity (stepthrough)

mcshaz committed:
more serializing tests (passing)

mcshaz committed:
valid string values clarified/expanded #24

mcshaz committed:
specifies UTC dates where spec mandates

mcshaz committed:
FreeBusy spec validation

mcshaz committed:
Failing FreeBusy serializationtests

mcshaz committed:
datestamp in UTC

mcshaz committed:
further test serialization - UTC T00:00 failing

mcshaz committed:
fix UTC midnight issue #77

mcshaz committed:
CalDateTime constructor comments for HasTime

mcshaz committed:
DateTimeValue typo

mcshaz committed:
merge rename .NET .Net

mcshaz committed:
spec validation project included

mcshaz committed:
compare 2 components - CalDateTime.HasTime should be equal

mcshaz committed:
rename file iCalDateTime to CalDateTime

rianjs commented on Ical.Net.nuspec:
```diff
@@ -2,16 +2,14 @@
     <metadata>
         <id>Ical.Net</id>
-        <version>2.2.5</version>
+        <version>2.1.5</version>
```
This doesn't seem right. I bumped the minor version number from 1 to 2, because I added a couple of public constants a few PRs ago. ical.net is using [semantic versioning](http://semver.org/):

> Given a version number MAJOR.MINOR.PATCH, increment the:
> 1. MAJOR version when you make incompatible API changes,
> 2. MINOR version when you add functionality in a backwards-compatible manner, and
> 3. PATCH version when you make backwards-compatible bug fixes.

rianjs commented on Ical.Net.nuspec:
```diff
@@ -2,16 +2,14 @@
         <description>An iCalendar (RFC 5545) control library for .NET. A fork and continuation of the original dday.ical library including bug fixes, and many performance enhancements.</description>
         <projectUrl>https://github.com/rianjs/ical.net</projectUrl>
         <licenseUrl>https://github.com/rianjs/ical.net/blob/master/license.md</licenseUrl>
-        <iconUrl>https://github.com/rianjs/ical.net/raw/master/logo.png</iconUrl>
```
If you merge this PR, you'll merge these deletions.

mcshaz replied:
Thanks for your patience & review as I get my head around Github – While I have used it in the past, it has only been for projects I am developing, and this leaves a number of lessons to be learned and improvements to be made. I will make sure deletions are not re-introduced to the project. 

Enjoy your sabbatical.

mcshaz committed:
date/time property explanations

mcshaz committed:
string constant access classifications in serialization tests

mcshaz committed:
test of assigning attendee list directly

This was broken in the old dday.ical. Test added to ensure this
frustrating error never returns

mcshaz committed:
rename to ICalendarOrgSpecificFormat + permission email

mcshaz committed:
vtimezone comments
